plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -875,4 +880,123 @@ CompletableFuture<GetObjectAttributesResponse> getBlobMetadata(S3AsyncClient s3A
 
         return SocketAccess.doPrivileged(() -> s3AsyncClient.getObjectAttributes(getObjectAttributesRequest));
     }
+
+    @Override
+    public void deleteAsync(ActionListener<DeleteResult> completionListener) {
```

Comment:
This revision looks good but may be in a follow up PR we can add a top level timeout in case due to a SDK bug if the application thread doesn't get released ever. I have seen these very rare cases while building and testing permit based remote transfer capability.